### PR TITLE
gr.jl

### PR DESCRIPTION
**Pull request by jheinen**
Still some TODOs and missing features – but moving ahead. See http://goo.gl/YaCMTf

### CODE REVIEW

File: src/backends/gr.jl [hunk 1]
```diff
@@ -217,8 +309,11 @@ end
 
 # ----------------------------------------------------------------
 
-function Base.writemime(io::IO, ::MIME"image/png", plt::PlottingObject{GRPackage})
-  # TODO: write a png to io
+function Base.writemime(io::IO, m::MIME"image/png", plt::PlottingObject{GRPackage})
+  isijulia() || return
```
Comment (tbreloff) [inline src/backends/gr.jl]: Is it only possible to write a png from IJulia?  I frequently want to do this from the REPL.

File: src/backends/supported.jl [hunk 2]
```diff
@@ -234,10 +234,11 @@ supportedArgs(::GRPackage) = [
   ]
 supportedAxes(::GRPackage) = _allAxes
 supportedTypes(::GRPackage) = [:none, :line, :path, :steppre, :steppost, :sticks,
-                                   :scatter, :heatmap, :hexbin, :hist, :density, :bar,
-                                   :hline, :vline, :contour, :path3d, :scatter3d, :surface, :wireframe]
+                               :scatter, :heatmap, :hexbin, :hist, :density, :bar,
+                               :hline, :vline, :contour, :path3d, :scatter3d, :surface,
+                               :wireframe, :ohlc, :pie]
```
Comment (tbreloff) [inline src/backends/supported.jl]: I thought you only support `[:none, :line, :scatter]` so far.  Why do you have so many listed?

File: src/backends/supported.jl [hunk 3]
```diff
@@ -234,10 +234,11 @@ supportedArgs(::GRPackage) = [
...
 supportedStyles(::GRPackage) = [:auto, :solid, :dash, :dot, :dashdot]
-supportedMarkers(::GRPackage) = [:none, :ellipse, :rect, :diamond, :utriangle, :dtriangle, :pentagon, :cross, :xcross, :star5]
+supportedMarkers(::GRPackage) = vcat([:none, :ellipse, :rect, :diamond, :utriangle, :dtriangle, :pentagon, :hexagon, :cross, :xcross, :star5], Shape)
```
Comment (tbreloff) [inline src/backends/supported.jl]: I don't think you support custom shapes yet, so you shouldn't add this.

**Comment by tbreloff** [comment]
This looks great... thanks! I put a few comments inline.  Would you like me to merge soon, or do you want to keep this PR open and just keep pushing new commits to it?

**Comment by jheinen** [comment]
Please let me keep pushing more commits - there are still too much missing features. I'll probably find some time next week ...

**Pull request closed without merging**